src/spawn.cpp

```diff
@@ -198,14 +200,17 @@ bool Spawn::isInSpawnZone(const Position& pos)
 bool Spawn::spawnMonster(uint32_t spawnId, MonsterType* mType, const Position& pos, Direction dir, bool startup /*= false*/)
 {
 	std::unique_ptr<Monster> monster_ptr(new Monster(mType));
-	if (startup) {
-		//No need to send out events to the surrounding since there is no one out there to listen!
-		if (!g_game.internalPlaceCreature(monster_ptr.get(), pos, true)) {
-			return false;
-		}
-	} else {
-		if (!g_game.placeCreature(monster_ptr.get(), pos, false, true)) {
-			return false;
+	bool result = g_events->eventMonsterOnSpawn(monster_ptr.get(), pos, startup, false);
```

Comment:
Is there any reason to not just return here?
```c++
if (!g_events->eventMonsterOnSpawn(monster_ptr.get(), pos, startup, false)) {
    return false;
}
```

Reply:
Like so?

```c++
bool Spawn::spawnMonster(uint32_t spawnId, MonsterType* mType, const Position& pos, Direction dir, bool startup /*= false*/)
{
	std::unique_ptr<Monster> monster_ptr(new Monster(mType));
	if (!g_events->eventMonsterOnSpawn(monster_ptr.get(), pos, startup, false)) {
		return false;
	} else {
		if (startup) {
			//No need to send out events to the surrounding since there is no one out there to listen!
			if (!g_game.internalPlaceCreature(monster_ptr.get(), pos, true)) {
				return false;
			}
		} else {
			if (!g_game.placeCreature(monster_ptr.get(), pos, false, true)) {
				return false;
			}
		}
	}

	Monster* monster = monster_ptr.release();
	monster->setDirection(dir);
	monster->setSpawn(this);
	monster->setMasterPos(pos);
	monster->incrementReferenceCounter();

	spawnedMap.insert(spawned_pair(spawnId, monster));
	spawnMap[spawnId].lastSpawn = OTSYS_TIME();
	return true;
}
```